File: backend/src/main/java/dev/agiro/fanel/chores/api/ChoresApi.java
package dev.agiro.fanel.chores.api;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

public interface ChoresApi {
    List<ChoreDto> list(UUID householdId);

    ChoreDto create(UUID householdId, String title, UUID assigneeId, LocalDate dueDate,
                    RecurrenceFrequency recurrenceFreq, Integer recurrenceInterval, List<UUID> rotationMemberIds);

    default ChoreDto create(UUID householdId, String title, UUID assigneeId) {
        return create(householdId, title, assigneeId, null, null, null, null);
    }

    /** Partial update of title/assignee/done; leaves the recurrence configuration untouched. Setting
     * {@code done = true} on a recurring chore rolls it to its next occurrence (due date advanced, assignee
     * rotated) instead of leaving it marked done; see {@code Chore#complete()}. */
    ChoreDto update(UUID householdId, UUID choreId, String title, UUID assigneeId, Boolean done);

    /** Replaces the chore's recurrence configuration outright. Pass {@code recurrenceFreq = null} to make the
     * chore non-recurring again. */
    ChoreDto updateRecurrence(UUID householdId, UUID choreId, LocalDate dueDate, RecurrenceFrequency recurrenceFreq,
                             Integer recurrenceInterval, List<UUID> rotationMemberIds);

    void delete(UUID householdId, UUID choreId);
}

File: android/app/src/main/java/dev/agiro/fanel/android/data/remote/ChoresApi.kt
package dev.agiro.fanel.android.data.remote

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface ChoresApi {
    @GET("/api/v1/households/{householdId}/chores")
    suspend fun list(@Path("householdId") householdId: String): List<ChoreDto>

    @POST("/api/v1/households/{householdId}/chores")
    suspend fun create(
        @Path("householdId") householdId: String,
        @Body request: CreateChoreRequest
    ): ChoreDto

    @PATCH("/api/v1/households/{householdId}/chores/{choreId}")
    suspend fun update(
        @Path("householdId") householdId: String,
        @Path("choreId") choreId: String,
        @Body request: UpdateChoreRequest
    ): ChoreDto

    @PUT("/api/v1/households/{householdId}/chores/{choreId}/recurrence")
    suspend fun updateRecurrence(
        @Path("householdId") householdId: String,
        @Path("choreId") choreId: String,
        @Body request: UpdateRecurrenceRequest
    ): ChoreDto

    @DELETE("/api/v1/households/{householdId}/chores/{choreId}")
    suspend fun delete(
        @Path("householdId") householdId: String,
        @Path("choreId") choreId: String
    )
}
